src/scheduler.c

```diff
@@ -581,9 +581,10 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
 
 void scheduler_delete_thread(jl_ptls_t ptls) JL_NOTSAFEPOINT
 {
-    if (jl_atomic_exchange_relaxed(&ptls->sleep_check_state, sleeping_like_the_dead) != sleeping) {
-        int wasrunning = jl_atomic_fetch_add_relaxed(&nrunning, -1);
```

Comment:
I had actually caught this ordering mistake in `rr`, with the `threads_exec` test from uv_library_shutdown, since we decremented nrunning here before the `jl_wakeup_thread` later

Reply:
hm, not actually it was a different spot on closer look than this data race ([add missing increment of nrunning for jl_adopt_thread](https://github.com/JuliaLang/julia/pull/45794/commits/ab1dda237fdabfd0fd38eb6c51ce219e5e15c1f4))